chapters/connectors.tex

```diff
@@ -903,26 +905,61 @@ \subsection{Connection Graphs and Their Operators}\label{overconstrained-equatio
 
 \subsection{Generation of Connection Graph Equations}\label{generation-of-connection-graph-equations}
 
+When generating connection graph equations, only the overdetermined components of a connector are considered.
+The connection graph equations replace the equality-equations for variables that are neither flow nor stream in \cref{generation-of-connection-equations}.
+
+\subsubsection{Remove Redundant Connect-Equations}
+In order to eliminate any redundant \lstinline!connect!-equation the following preparation is needed.
+\begin{nonnormative}
+In the common case where there are no redundant \lstinline!connect!-equations a \lstinline!connect! between connectors with \emph{one} overdetermined component will directly generate \emph{one} optional spanning-tree edge.
+\end{nonnormative}
+\begin{enumerate}
+\item
+  The connection sets are built similarly to the normal way, but keeping the overdetermined components as primitives.
+\item
+  Instead of generating the equality-equation for an overdetermined component, an optional spanning-tree edge in the virtual connection graph is constructed.
```

Comment:
I feel like this does not quite reflect what we discussed during the meeting: we consider connect equation redundant if it results in the same connection graph. Or at least, not as I understand it. Here is an example to discuss. Let's assume we have 4 connectors a, b, c, d that have some overdetermined types in them (R). Assume that we have:
```
Connections.root(a.R);
Connections.branch(a.R,b.R);
connect(a,b);
connect(a,c);
connect(a,d);
connect(b,c);
```
According to the rules above, we create a connection set based only on connect equations.
**Question one**: Shouldn't we include connectors connected by branches into the connection set as well? Assume, there was no connect(b,c) in that set. We have 4 connectors and 3 connect equations and they are all needed to form the same connection set, so no redundant connections. But then connect(a,b) is an optional edge in the graph but that is not in the tree, and we would need to generate the equalityConstraint for it. Should connect(a,b) be treated as redundant or should it generate this equalityConstraint?

We follow the rules above and construct the connection set based only on connect equations, we get: {a,b,c,d}. When we generate equality-equations for non-flow connectors, it doesn't matter what the original equations were since it is equality, so we are free to generate: a.t = b.t, b.t = c.t, c.t = d.t, assuming t is some non-flow, non-overdetermined type. The text above implies to me that we should follow the same logic for the overdetermined connectors but instead generate branches of the tree. So we could generate: [a.R,b.R], [b.R,c.R], [c.R,d.R].
**Question two**: Does it matter if I end up with an edge that doesn't correspond to any connect-equation like [c.R,d.R]?
The resulting equations would be: a.R = b.R (from the branch(...)), b.R = c.R, c.R = d.R, equalityConstrant(a.R,b.R). Is that correct?

**Question three**: Assume we had no branch(a.R,b.R). The connection set would be the same, the edges for the connection graph would be as before: [a.R,b.R], [b.R,c.R], [c.R,d.R]. But now there is no branch to define an additional unbreakable edge, so our graph is also a tree. The resulting equations would be a.R = b.R, b.R = c.R, c.R = d.R. No equalityConstraint needs to be introduced for connect(a,d). Is that correct?

This new formulation is very different from my previous understanding of how connections are elaborated. Maybe some differences do not matter at the end, because the end set of equations is equivalent, but I want to make sure that it is indeed so.

Reply:
I will have to consider whether the first two questions are relevant later; and/or whether we should guard against it in some way. But I don't think it should block this PR.

To me the two steps are separate (so that Connections.branch will not matter for the redundancy), and the model is almost guaranteed to fail.

The basic issue is that having both `connect(a, b)` and `Connections.branch(a.R, b.R)` doesn't make sense to me, since `Connections.branch(a.R, b.R)` should be used in combination with equations in the model (this is stated non-normatively in the chapter).

A typical case is Modelica.Mechanics.MultiBody.Parts.FixedTranslation where:
```
  Connections.branch(frame_a.R, frame_b.R);
  frame_b.r_0 = frame_a.r_0 + Modelica.Mechanics.MultiBody.Frames.resolve1(
    frame_a.R, r);
  frame_b.R = frame_a.R;
```
These three equations form an equationy unit linking `frame_a` to `frame_b`. Thus I don't see how we can also have a normal connect-equation between them inside the same model. (If they are connected from the outside, it would be more complicated.)

It's like having a block with:
```
  Modelica.Blocks.Interfaces.RealInput u;
  Modelica.Blocks.Interfaces.RealOutput Real y;
equation
  connect(u, y);
   y=k*u;
```
You can write some model like that that works - but I don't think it is good modeling.  (Well, just having `Real k;` and indirectly solving for it works.)

I see two solutions for these:
- Ignore it; as it is similar for normal connectors.
- Forbid having Connection.branch overlapping with connect inside one model; but I believe that can be a separate PR.

-
For the last question: yes, it would just be a tree without equalityConstraint.